creator-node/src/routes/userReplicaSet.js

```diff
@@ -0,0 +1,197 @@
+     */
+    let requesterWalletRecoveryObj = { spID, reqTimestamp }
+    let recoveredDelegateOwnerWallet = (recoverWallet(requesterWalletRecoveryObj, reqSignature)).toLowerCase()
+    if (delegateOwnerWalletFromSPFactory !== recoveredDelegateOwnerWallet) {
+      return errorResponseBadRequest(
+        'Request for proposal must be signed by delegate owner wallet registered on L1 for spID'
+      )
+    }
+
+    /**
+     * Request node's up-to-date health info
+     *
+     * - uses endpoint registered on L1 for spID
+     * - passes `randomBytesToSign` string in request to check that response was signed for provided data
+     */
+    const randomBytesToSign = (await randomBytes(18)).toString()
+    let nodeHealthCheckResp = await axios({
+      baseURL: nodeEndpointFromChain,
+      url: '/health_check',
+      method: 'get',
+      timeout: 1000,
+      params: {
+        randomBytesToSign
+      }
+    })
+    nodeHealthCheckResp = parseCNodeResponse(
+      nodeHealthCheckResp,
+      ['healthy', 'creatorNodeEndpoint', 'spID', 'spOwnerWallet', 'randomBytesToSign']
+    )
+
+    /**
+     * Confirm health check returns healthy and response data matches on-chain data
+     */
+    if (
+      !(nodeHealthCheckResp.healthy)
+      || (nodeHealthCheckResp.creatorNodeEndpoint !== nodeEndpointFromChain)
+      || (nodeHealthCheckResp.spID !== spID)
+      || ((nodeHealthCheckResp.spOwnerWallet).toLowerCase() !== ownerWalletFromChain.toLowerCase())
+    ) {
+      return errorResponseServerError(`CN unhealthy or misconfigured`)
+    }
+
+    /**
+     * Confirm health check response was signed by delegate owner wallet registered on L1
+     *    for spID and includes `randomBytesToSign`
+     */
+    let {
+      timestamp: respTimestamp,
+      signature: respSignature,
+      ...responseData
+    } = nodeHealthCheckResp.rawResponse
+    const responderWalletRecoveryObj = { ...responseData, randomBytesToSign, timestamp: respTimestamp }
+    recoveredDelegateOwnerWallet = (recoverWallet(responderWalletRecoveryObj, respSignature)).toLowerCase()
+    if (delegateOwnerWalletFromSPFactory !== recoveredDelegateOwnerWallet) {
+      return errorResponseBadRequest(
+        'Health check response must be signed by delegate owner wallet registered on L1 for spID'
+      )
+    }
+
+    // Generate proposal signature artifacts
+    const proposalSignatureInfo = await libs.contracts.UserReplicaSetManagerClient.getProposeAddOrUpdateContentNodeRequestData(
+      spID,
+      delegateOwnerWalletFromSPFactory,
+      config.get('spID'),
+      config.get('delegateOwnerWallet'),
+      libs.ethWeb3Manager.getWeb3()
+    )
+
+    return successResponse({
+      nonce: proposalSignatureInfo.nonce,
+      sig: proposalSignatureInfo.sig,
+      spID: config.get('spID')
+    })
+  }))
+
+  app.get('/submitRFP', handleResponse(async (req, res, next) => {
```

Comment:
why should this be a route? imagine a node coming up, requesting 3 proposer signatures - when does it have to request the node to addOrUpdate

Reply:
yeah this def won't be a route - thats why i mentioned in my msg to only look at the one `ursm_request_for_proposal` route since everything else is just testing code